ccx/api/datastore_create.go

```diff
@@ -123,61 +122,63 @@ func DatastoreFromResponse(r DatastoreResponse) ccx.Datastore {
 		CloudProvider:     r.CloudProvider,
 		CloudRegion:       r.Region,
 		InstanceSize:      r.InstanceSize,
-		VolumeType:        pointers.String(r.DiskType),
-		VolumeSize:        int64(pointers.Uint64(r.DiskSize)),
-		VolumeIOPS:        int64(pointers.Uint64(r.IOPS)),
+		VolumeType:        pString(r.DiskType),
+		VolumeSize:        int64(pUint64(r.DiskSize)),
+		VolumeIOPS:        int64(pUint64(r.IOPS)),
 		NetworkType:       "", // todo
 		HAEnabled:         r.HighAvailability,
-		VpcUUID:           pointers.String(r.VpcUUID),
+		VpcUUID:           pString(r.VpcUUID),
 		AvailabilityZones: r.AZS,
 	}
 }
 
-// Create a new datastore
-func (cli *Client) Create(ctx context.Context, c ccx.Datastore) (*ccx.Datastore, error) {
+func (svc *DatastoreService) Create(ctx context.Context, c ccx.Datastore) (*ccx.Datastore, error) {
 	cr := CreateRequestFromDatastore(c)
 
 	var b bytes.Buffer
 	if err := json.NewEncoder(&b).Encode(cr); err != nil {
 		return nil, errors.Join(ccx.RequestEncodingErr, err)
 	}
 
-	url := cli.conn.BaseURL + "/api/prov/api/v2/cluster"
+	url := svc.baseURL + "/api/prov/api/v2/cluster"
 	req, err := http.NewRequest(http.MethodPost, url, &b)
 	if err != nil {
-		return nil, errors.Join(ccx.RequestInitializationErr, err)
+		return nil, errors.Join(ccx.CreateFailedErr, ccx.RequestInitializationErr, err)
 	}
 
-	token, err := cli.auth.Auth(ctx)
+	token, err := svc.auth.Auth(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Authorization", token)
-	client := &http.Client{Timeout: cli.conn.Timeout}
+	client := &http.Client{Timeout: ccx.DefaultTimeout}
 
 	res, err := client.Do(req)
 	if err != nil {
 		return nil, errors.Join(ccx.RequestSendingErr, err)
 	}
 
 	if res.StatusCode == http.StatusBadRequest {
-		return nil, chttp.ErrorFromErrorResponse(res.Body)
+		return nil, fmt.Errorf("%w: %w", ccx.CreateFailedErr, lib.ErrorFromErrorResponse(res.Body))
 	}
 
 	if res.StatusCode != http.StatusCreated {
-		return nil, fmt.Errorf("%w: status = %d", chttp.ErrorFromErrorResponse(res.Body), res.StatusCode)
+		return nil, fmt.Errorf(":%w :%w: status = %d", ccx.CreateFailedErr, lib.ErrorFromErrorResponse(res.Body), res.StatusCode)
 	}
 
 	var rs DatastoreResponse
-	if err := chttp.DecodeJsonInto(res.Body, &rs); err != nil {
-		return nil, err
+	if err := lib.DecodeJsonInto(res.Body, &rs); err != nil {
+		return nil, fmt.Errorf("%w: %w", ccx.CreateFailedErr, err)
 	}
 
 	newDatastore := DatastoreFromResponse(rs)
 
-	if err := cli.LoadAll(ctx); err != nil {
-		return nil, errors.Join(ccx.ResourcesLoadFailedErr, err)
+	status, err := svc.jobs.Await(ctx, rs.UUID, deployStoreJob, svc.timeout)
+	if err != nil {
+		return nil, fmt.Errorf("%w: awaiting deploy job: %w", ccx.CreateFailedErr, err)
```

Comment:
If something fails or times out, we throw away the store ID?

Reply:
unfortunately we have to give up at some point :disappointed:

Reply:
Isn't it possible to keep the ID, and record that the thing is in an unknown state? Otherwise we risk making lots of broken stores?

Reply:
In [this commit](https://github.com/severalnines/terraform-provider-ccx/pull/19/commits/9df8ac333f4ef5b13e34a212f3c2107fa2240d9b) if create is without error but await / read fails, the error `CreateFailedReadErr` is returned.

For this error, the id is set in terraform.